src/Core/System.php

```diff
@@ -236,10 +236,27 @@ public static function currentLoad()
 		return max($load_arr[0], $load_arr[1]);
 	}
 
+	/**
+	 * Redirects to another URL and exits this process.
+	 *
+	 * @param string $toUrl The destination URL (Default is empty, which is the default page of the Friendica node)
+	 * @param bool $ssl if true, base URL will try to get called with https:// (works just for relative paths)
+	 */
+	public static function redirect($toUrl = '', $ssl = false)
```

Comment:
Why to ```App```? Currently we have static classes in ```System``` and non static ones in ```App```.

Reply:
Yes, and `goaway()` depends on `App->getBaseUrl()` which isn't static, so it doesn't make sense to keep it static.

Reply:
We could use ```self::baseUrl()``` as well :-)

Reply:
Yes but `self::baseUrl()` is just a shim for `App->getBaseUrl()`!

Reply:
This had been an example. I think we really should hold all static functions in ```System``` while the other ones should be in ```App```. And I think it doesn't matter what is done behind. Classes are blackboxes.

Reply:
I don't know if System::redirect is therefor really necessary, because it encapsulates "just" a header call and exit.

On the other hand I will make some tests more easy (and we can enable some of the ignore API tests) ;-)

Reply:
It isn't "really necessary", just slightly better (one-line call instead of two) when we don't need the `App` context for an external URL redirection.

Reply:
I was currently reworking it .. But I think I stop it for now and wait until we all have the same view.. Because it's refactoring through the whole repo everytime ;-) ;-)

So we have to differ when we use a full URL and when do we redirect from a relative path. Doesn't sound bad, but we have to add at System::redirect that it should not get used like `System::redirect($a->getBaseURL() . 'realtive/path');`. Otherwise both versions will get used for sure ;-)

Reply:
Being mindful whether we need the `App` context or not is indeed more complicated. The good news is that we totally control the software, so we can ensure `System::redirect($a->getBaseURL() . '/relative/path');` isn't used even if it's valid.

This idea of separating internal/external redirects is actually growing on me. That's one `if` less!

Reply:
How can we ensure that `System::redirect($a->getBaseURL() . '/relative/path');` isn't used? You mean because of the reviews?

I know situations where we have f.e.
```php
$url = $record['url'];
System::redirect($url);
```

So we need to know if we record a full or a relative path. I don't know if we don't oversee it in some cases just because it isn't clear at first sight.

Reply:
> You mean because of the reviews?

Yes. and even then, it wouldn't be that big of a deal, just inelegant.

> So we need to know if we record a full or a relative path.

If you look into the DB, you'll see if we use relative of absolute URLs, and also we don't have to do the transition in one PR.

Reply:
But if it's not that big of a deal, we can live with the `System::redirect()` for both variants.
I think it's more bulletproof. 

The worst thing we can have is that we oversee a statement like 
```php
$url = 'https://some.other.url/'
[...]
$a->redirect($url);
```
because this would lead to unexpected errors.

Reply:
So you favour ```BaseObject::getApp->redirect(...)```? (Which would be the alternate solution when you haven't got an ```App``` class variable at hand)

Reply:
No, I favor my solution above, but I will begrudgingly accept this alternative, because we shouldn't do any redirect outside of Modules where App is available anyway. Any ugly use like this outside of modules should be replaced with something else.

Reply:
What about splitting them as said and adding an exception/logger entry if somebody calls `$a->redirect()` with a full url? Just an idea..

Reply:
Oh, I didn't think about exceptions. This is radical but it would work for me.

Reply:
I don't see the need for an exception there - and BTW: Why shouldn't one use a complete URL there?

Reply:
I know it's somewhat a novel concept for you, but knowing what kind of data you expect is important in programming.

If we have a complete URL, then we don't need the `App` context. If we have a relative URL, then we do. And if we have a complete URL where we expect a relative URL means that something is wrong somewhere else in the code. The opposite is true, if we expect an absolute URL we shouldn't handle relative URLs as well.

Reply:
But I think we are using this function as well for external redirects. So do you want to have two different redirect functions? One for internal, one for external?

Reply:
Exactly. Instead of having one catchall function which doesn't care if the calling logic is sound or not, we'll have two purposeful methods that will check for strong parameter type.

Reply:
I don't see a reason why, I must confess. With the same explanation you could create a redirect for URL with or without URL parameters.

Reply:
Before: "No http(s) protocol? Bah, it must be a relative URL, let's prepend the base URL."
After: "No http(s) protocol? Uh-oh, I'm only expecting absolute URLs, something's wrong."

It won't solve all our redirection issues, but it will catch the ones that are related to type mismatch.

Reply:
The point isn't in the URL check, the point is "Why did we get a relative URL when we are expecting an absolute one, there must be something wrong with the calling code."

The redirect method is simple enough that it's bug-proof, but the calling code may not be.

Reply:
Okay. So how do we name these functions so that everyone knows which one is for what?

Reply:
`App->internalRedirect()` and `System::externalRedirect`? Interestingly, the former would be calling the latter.

Or `App->relativeRedirect()` and `System::absoluteRedirect`? It's more accurate but less intuitive.

Reply:
On my way :D